core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1621,8 +1621,31 @@ public UnifyResult apply(UnifyRuleCall call) {
             }
           }
         }
+
+        // try to adjust RelDataType
+        List<RexNode> projectExprs = new ArrayList<>();
```

Comment:
Use `final` ?

Reply:
Thanks for you review.  create `createProjects` to adjust RelDataType.